.github/workflows/nodejs.yml

```diff
@@ -195,6 +195,71 @@ jobs:
         if: always()
         run: docker stop rippled-service
 
+  generate-documentation:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [18.x]
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Setup npm version 10
+        run: |
+          npm i -g npm@10 --registry=https://registry.npmjs.org
+
+      - name: Cache node modules
+        id: cache-nodemodules
+        uses: actions/cache@v4
+        env:
+          cache-name: cache-node-modules
+        with:
+          # caching node_modules
+          path: |
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-deps-${{ matrix.node-version }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-deps-${{ matrix.node-version }}-
+
+      - name: Install Dependencies
+        if: steps.cache-nodemodules.outputs.cache-hit != 'true'
+        run: npm ci
+
+      - run: npm run build
+
+      - name: Generate Documentation
+        run: npm run docgen
+
+      - name: Upload documentation files as artifact
+        id: docs-artifact
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: docs/
+          retention-days: 10
+
+  deploy-docs-pages:
+    permissions:
+      id-token: write  # Needed for OIDC authentication
+      pages: write # this permission is needed for deploying into Github Pages
+    environment:
+      name: github-pages
+      url: ${{ steps.docs-artifact.outputs.page_url }}
+    runs-on: ubuntu-latest
+    needs: generate-documentation
+    # Deploy docs only pushes into the main branch
+    if: success() && github.ref == 'refs/heads/main'
```

Comment:
This should be run on a new published version, not on every commit

Reply:
1. On the [documentation website](https://js.xrpl.org/), I don't observe a menu of various versions. Usually, softwares allow users to select a version before viewing the relevant docs. Does this need to be configure for Typedoc/Github Pages?

2. I agree with your point:
> This should be run on a new published (non-beta) version, not on every commit

However, is there any harm in publishing the docs on every commit? (Provided we specify the necessary disclaimers to the users) ?

Especially, if we don't have the infrastructure to render versioned-documentation, I believe its good to keep the docs fresh.

Reply:
> This should be run on a new published (non-beta) version, not on every commit

If you feel strongly about it, I can accomplish this with modifications to the if-condition.

Reply:
> However, is there any harm in publishing the docs on every commit? (Provided we specify the necessary disclaimers to the users) ?

It'll be a lot more confusing if users think there's a feature out because the documentation says so, but it's not released.

We don't necessarily need to have the infra for versioned documentation (though we could try to set something up with readthedocs), but the trigger of this action could just be changed to be on a non-beta tag instead, for example.

Reply:
ok. updated in [70d8ced](https://github.com/XRPLF/xrpl.js/pull/2954/commits/70d8ceda6df21d811bd6143ebd66db12d461ad90)